google-http-client/src/main/java/com/google/api/client/http/HttpResponse.java

```diff
@@ -334,11 +334,11 @@ public InputStream getContent() throws IOException {
         try {
           // gzip encoding (wrap content with GZipInputStream)
           if (!returnRawInputStream
-                  && this.contentEncoding != null
-                  && (CONTENT_ENCODING_GZIP.equalsIgnoreCase(this.contentEncoding.trim())
-                      || CONTENT_ENCODING_XGZIP.equalsIgnoreCase(this.contentEncoding.trim()))) {
+              && this.contentEncoding != null
+              && (CONTENT_ENCODING_GZIP.equals(this.contentEncoding.trim())
```

Comment:
I think per spec you need ignore case--correct me if I'm wrong about that--but you need to do it with the right locale, not necessarily the default.

Reply:
The header name is case-insensitive, header values are dependent on which field. For content encoding, valid values are a list of defined tokens [maintained by IANA](http://www.iana.org/assignments/http-parameters/http-parameters.xhtml#content-coding).

Reply:
@elharo , @chingor13 - I have another commit that has case-insensitive checks and hopefully addresses the prior locale issues mentioned. But given the recent point raised by @chingor13 , it seems that only the header NAME should be case-insensitive and not the actual values which I believe the current PR addresses.

Let me know if you'd like me to push the new commit to the branch.